File: Python/Math-Array-String/76.py
# 2351. First Letter to Appear Twice

class Solution:
    def repeatedCharacter(self, s: str) -> str:
        seen = set()
        for char in s:
            if char in seen:
                return char
            seen.add(char)
        return ""  # This won't be reached as per the problem constraints

sol = Solution()
output = sol.repeatedCharacter("abccbaacz")
print(output)  # Output: c

File: JavaScript/Math-Array-String/76.js
// 2351. First Letter to Appear Twice


/**
 * @param {string} s
 * @return {character}
 */
var repeatedCharacter = function(s) {
    let seen = new Set();
    
    for (let char of s) {
        if (seen.has(char)) {
            return char;
        }
        seen.add(char);
    }
    
    return ""; // This won't be reached as per problem constraints
};

// Example usage:
console.log(repeatedCharacter("abccbaacz")); // Output: "c"
console.log(repeatedCharacter("abcdd"));    // Output: "d"
